### ISSUE TITLE
Exception in 1.12.2 Sponge 7.1.0-BETA

### ISSUE THREAD

**Issue by ImagePrototype** [JimiIT92/UniverseGuard2#9]
When is the upgrade to Sponge API 7?
```
java.lang.NoClassDefFoundError: org/spongepowered/api/event/cause/entity/spawn/EntitySpawnCause
        at java.lang.Class.getDeclaredMethods0(Native Method) ~[?:1.8.0_121]
        at java.lang.Class.privateGetDeclaredMethods(Unknown Source) ~[?:1.8.0_121]
        at java.lang.Class.privateGetPublicMethods(Unknown Source) ~[?:1.8.0_121]
        at java.lang.Class.getMethods(Unknown Source) ~[?:1.8.0_121]
        at org.spongepowered.common.event.SpongeEventManager.registerListener(SpongeEventManager.java:226) ~[SpongeEventManager.class:1.12.2-2555-7.1.0-BETA-2825]
        at org.spongepowered.common.event.SpongeEventManager.registerListeners(SpongeEventManager.java:293) ~[SpongeEventManager.class:1.12.2-2555-7.1.0-BETA-2825]
        at com.universeguard.utils.EventUtils.registerEvent(EventUtils.java:38) ~[EventUtils.class:?]
        at com.universeguard.UniverseGuard.registerEvents(UniverseGuard.java:308) ~[UniverseGuard.class:?]
        at com.universeguard.UniverseGuard.onInit(UniverseGuard.java:136) ~[UniverseGuard.class:?]
        at org.spongepowered.common.event.listener.GameInitializationEventListener_UniverseGuard_onInit11.handle(Unknown Source) ~[?:?]
        at org.spongepowered.common.event.RegisteredListener.handle(RegisteredListener.java:95) ~[RegisteredListener.class:1.12.2-2555-7.1.0-BETA-2825]
        at org.spongepowered.mod.event.SpongeModEventManager.post(SpongeModEventManager.java:338) [SpongeModEventManager.class:1.12.2-2555-7.1.0-BETA-2825]
        at org.spongepowered.mod.event.SpongeModEventManager.post(SpongeModEventManager.java:371) [SpongeModEventManager.class:1.12.2-2555-7.1.0-BETA-2825]
        at org.spongepowered.common.SpongeImpl.postEvent(SpongeImpl.java:213) [SpongeImpl.class:1.12.2-2555-7.1.0-BETA-2825]
        at org.spongepowered.mod.SpongeMod.onStateEvent(SpongeMod.java:232) [SpongeMod.class:1.12.2-2555-7.1.0-BETA-2825]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_121]
        at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_121]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_121]
        at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_121]
        at com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Subscriber.java:91) [minecraft_server.1.12.2.jar:?]
        at com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Subscriber.java:150) [minecraft_server.1.12.2.jar:?]
        at com.google.common.eventbus.Subscriber$1.run(Subscriber.java:76) [minecraft_server.1.12.2.jar:?]
        at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:399) [minecraft_server.1.12.2.jar:?]
        at com.google.common.eventbus.Subscriber.dispatchEvent(Subscriber.java:71) [minecraft_server.1.12.2.jar:?]
        at com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Dispatcher.java:116) [minecraft_server.1.12.2.jar:?]
        at com.google.common.eventbus.EventBus.post(EventBus.java:217) [minecraft_server.1.12.2.jar:?]
        at net.minecraftforge.fml.common.LoadController.sendEventToModContainer(LoadController.java:253) [LoadController.class:?]
        at net.minecraftforge.fml.common.LoadController.propogateStateMessage(LoadController.java:231) [LoadController.class:?]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_121]
        at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_121]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_121]
        at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_121]
        at com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Subscriber.java:91) [minecraft_server.1.12.2.jar:?]
        at com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Subscriber.java:150) [minecraft_server.1.12.2.jar:?]
        at com.google.common.eventbus.Subscriber$1.run(Subscriber.java:76) [minecraft_server.1.12.2.jar:?]
        at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:399) [minecraft_server.1.12.2.jar:?]
        at com.google.common.eventbus.Subscriber.dispatchEvent(Subscriber.java:71) [minecraft_server.1.12.2.jar:?]
        at com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Dispatcher.java:116) [minecraft_server.1.12.2.jar:?]
        at com.google.common.eventbus.EventBus.post(EventBus.java:217) [minecraft_server.1.12.2.jar:?]
        at net.minecraftforge.fml.common.LoadController.redirect$onPost$zza000(LoadController.java:560) [LoadController.class:?]
        at net.minecraftforge.fml.common.LoadController.distributeStateMessage(LoadController.java:148) [LoadController.class:?]
        at net.minecraftforge.fml.common.Loader.initializeMods(Loader.java:719) [Loader.class:?]
        at net.minecraftforge.fml.server.FMLServerHandler.finishServerLoading(FMLServerHandler.java:107) [FMLServerHandler.class:?]
        at net.minecraftforge.fml.common.FMLCommonHandler.onServerStarted(FMLCommonHandler.java:336) [FMLCommonHandler.class:?]
        at net.minecraft.server.dedicated.DedicatedServer.func_71197_b(DedicatedServer.java:219) [nz.class:?]
        at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:484) [MinecraftServer.class:?]
        at java.lang.Thread.run(Unknown Source) [?:1.8.0_121]
Caused by: java.lang.ClassNotFoundException: org.spongepowered.api.event.cause.entity.spawn.EntitySpawnCause
        at net.minecraft.launchwrapper.LaunchClassLoader.findClass(LaunchClassLoader.java:191) ~[launchwrapper-1.12.jar:?]
        at java.lang.ClassLoader.loadClass(Unknown Source) ~[?:1.8.0_121]
        at java.lang.ClassLoader.loadClass(Unknown Source) ~[?:1.8.0_121]
        ... 47 more
```

**Comment by ImagePrototype** [comment]
I have already read other issues, I understood that the problem is being solved. We are looking forward to UniverseGuard for Sponge API 7.

**Comment by JimiIT92** [comment]
Released API7 version for MC-1.12.2. Read more at [https://forums.spongepowered.org/t/universe-guard-2-an-easy-to-use-world-protection-plugin/21661/87](https://forums.spongepowered.org/t/universe-guard-2-an-easy-to-use-world-protection-plugin/21661/87)
